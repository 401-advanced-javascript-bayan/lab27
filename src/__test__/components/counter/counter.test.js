/* eslint-disable indent */
/* eslint-disable no-unused-vars */
/* eslint-disable react/no-unknown-property */
/* eslint-disable comma-dangle */
/* eslint-disable semi */
import React from 'react';
import Enzyme, { shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import Counter from '../../../components/counter.js'
Enzyme.configure({ adapter: new Adapter() });
describe('Testing our Counter component', () => {
    it('exists at the start of the app', () => {
        let app = shallow(<Counter />);
        expect(app.find('h3').exists()).toBeTruthy();
    })
    it('Counter renders', () => {
        let app = mount(<Counter />);
        let button = app.find('button');
        expect(button).toBeDefined();
        expect(app.state('ctr')).toBe(0);
    });
      
});