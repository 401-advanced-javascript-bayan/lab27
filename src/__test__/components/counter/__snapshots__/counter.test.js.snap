// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<counter/> renders correctly 1`] = `
<section>
  <h3>
    0
  </h3>
  <button
    onClick={[Function]}
  >
    click for counter
  </button>
  <button
    onClick={[Function]}
  >
    click for counter
  </button>
</section>
`;

exports[`Testing our Counter component render correctly to the DOM 1`] = `
<section>
  <h3>
    0
  </h3>
  <button
    className ="one"
    onClick={[Function]}
  >
    n+1
  </button>
  <button
    className ="two"
    onClick={[Function]}
  >
    n-1
  </button>
</section>
`;
